server/test/controllers/BlocksControllerSpec.scala

```diff
@@ -397,4 +428,25 @@ class BlocksControllerSpec extends MyAPISpec {
     (jsonBlock \ "previousBlockhash").asOpt[Blockhash] mustEqual block.previousBlockhash
     (jsonBlock \ "nextBlockhash").asOpt[Blockhash] mustEqual block.nextBlockhash
   }
+
+  private def matchBlockHeader(expected: BlockHeader, actual: JsValue) = {
+    val jsonBlockHeader = actual
+    val block = expected
+    val jsonFilter = (jsonBlockHeader \ "filter").as[JsValue]
+
+    (jsonBlockHeader \ "hash").as[Blockhash] mustEqual block.hash
+    (jsonBlockHeader \ "height").as[Height] mustEqual block.height
+    (jsonBlockHeader \ "time").as[Long] mustEqual block.time
+    (jsonBlockHeader \ "merkleRoot").as[Blockhash] mustEqual block.merkleRoot
+    (jsonBlockHeader \ "previousBlockhash").asOpt[Blockhash] mustEqual block.previousBlockhash
+
+    block match {
+      case BlockHeader.HasFilter(_, _, _, _, _, filter) => {
```

Comment:
If the expected header doesn't have a filter, this method will throw an exception, you should have seen a warning about that while compiling.